lib/controller/api_catalog.es6.js

```diff
@@ -7,20 +7,86 @@ require('angular');
 
 require('../compat.es6.js');
 require('../component/catalog_table.es6.js');
+require('../dao/dao_container.es6.js');
+require('../u2/ActionView.es6.js');
+require('../u2/DAOController.es6.js');
+require('../u2/DAOControllerView.es6.js');
+require('../u2/ScrollDAOTable.es6.js');
+require('../u2/SearchView.es6.js');
+require('../u2/ToggleListElementsView.es6.js');
+require('../web_apis/api_compat_data.es6.js');
+require('../web_apis/api_compat_data.es6.js');
+require('../web_apis/relational_to_compat.es6.js');
 const pkg = org.chromium.apis.web;
 
 angular.module('confluence').controller('catalogController', ['api',
   function(api) {
     const compatClassURL = `${window.location.origin}/${pkg.DAOContainer.COMPAT_MODEL_FILE_NAME}`;
+    let CompatData;
     pkg.ClassGenerator.create({
       classURL: compatClassURL,
-    }).generateClass().then(CompatData => {
+    }).generateClass().then(Cls => {
+      CompatData = Cls;
+
       const compatDAO = api.getApiServiceDAO(
           pkg.DAOContainer.COMPAT_NAME, CompatData);
+      const unorderdProperties = 
+          CompatData.getAxiomsByClass(pkg.CompatProperty);
+      const propDAO = foam.dao.ArrayDAO.create({array: unorderdProperties});
+      const E = foam.mlang.ExpressionsSingleton.create();
+      const composeReleaseOrder = function() {
+        let props = Array.from(arguments);
+        let order = E.DOT(pkg.CompatProperty.RELEASE, props.shift());
+        for (const prop of props) {
+          order = E.THEN_BY(order, E.DOT(pkg.CompatProperty.RELEASE, prop));
+        }
+        return order;
+      };
+
+      const orderedDAO = propDAO.orderBy(composeReleaseOrder(
+          pkg.Release.RELEASE_DATE,
+          pkg.Release.BROWSER_NAME,
+          pkg.Release.BROWSER_VERSION,
+          pkg.Release.OS_NAME,
+          pkg.Release.OS_VERSION));
+
+      return Promise.all([
+        orderedDAO.select(), 
+        orderedDAO.select(E.GROUP_BY(
```

Comment:
nit: Weird wrapping. better:

    orderedDAO.select(
      E.GROUP_BY(E.DOT(... ,
      foam.dao.ArraySink...

Reply:
Yeah. It's weird. I dropped it anyway. Just discovered a bug: order for default selected (by `browserName`) doesn't match order in browser selection list (which it should). I've implemented a different strategy that no longer uses DAO operations to cherry-pick the latest releases.